discovery-provider/plugins/notifications/src/sns.ts

```diff
@@ -117,25 +118,48 @@ type Device = {
 type Message = { title: string; body: string; data: object }
 
 export const sendPushNotification = async (
+  identityDb: Knex,
```

Comment:
i'm not a huge fan of this new dependency between sns and identity.. ideally sns is only handling sending the notif and the push notif class would handle writing to identity once it sees an endpoint is disabled. thoughts? ik you already made a bunch of changes to enable this so if you have a strong case we can keep as is

Reply:
oh you mean catching the error in the push notification class and doing the fix there?

Reply:
yea i think so, like maybe sns.sendPushNotification() returns false if it fails, and the push notif writes to identity.. something vaguely like:
```
const sentPushNotif = sns.sendPushNotification(...)
if (!sentPushNotif) 
    await identityDb('NotificationDeviceTokens')
          .where('awsARN', '=', device.targetARN)
          .update('enabled', false)
```

Reply:
Got it! Okay, ill do that